cli/gas.c

```diff
@@ -71,17 +77,18 @@ static void hexdump_data(struct switchtec_dev *dev, void __gas *map,
 	unsigned long addr = 0;
 	size_t bytes;
 	int last_match = 0;
+	int err;
 
 	while (map_size) {
 		if (is_alive && !is_alive())
 			return;
 
 		bytes = map_size > sizeof(line) ? sizeof(line) : map_size;
-		memcpy_from_gas(dev, line, map, bytes);
+		err = memcpy_from_gas(dev, line, map, bytes);
```

Comment:
This is where we should have `if (errno == EPERM)` print an error and exit.

And any other error that indicates no GAS can be read.

Reply:
Updated below.